### DISCUSSION TITLE
Material property 'strain', requested by 'TensileStrainEnergy' is not defined on block 0

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#23345] [Q&A Modules: Solid mechanics] [answered]
I need to define a custom AuxKernel to help me calculate the tensile strain part from a total strain tensor. But when I execute the simulation it reports this error. By looking at this [discussion](https://groups.google.com/g/moose-users/c/9ER_OSnvYRc/m/auxB_ISuBQAJ), I know the reason is that the block name is not the same as material property name. So what is the material_property name that ComputeSmallStrain declares? The following part is the corresponding part in my input file:
```
[Variables]
  [disp_x]
    order = FIRST
    family = LAGRANGE
  []
  [disp_y]
    order = FIRST
    family = LAGRANGE
  []
[]

[AuxVariables]
    [tensile_strain_energy]
        family = LAGRANGE
        order = FIRST
    []
[]

[Kernels]
  [TensorMechanics]
    #Stress divergence kernels
    displacements = 'disp_x disp_y'
   []
[]

[AuxKernels]
    [TensileStrainEnergy]
        type = TensileStrainEnergyAux
        rank_two_tensor = strain
        variable = tensile_strain_energy
        lambda = 120
        mu = 80
        block = 0
    []
[]
[Materials]
  [elasticity]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 1e9
    poissons_ratio = 0.3
    block = 0
  []
  [./strain]
    type = ComputeSmallStrain
    displacements = 'disp_x disp_y'
    block = 0
  [../]
  [./stress]
    type = ComputeLinearElasticStress
    block = 0
  [../]
[]
```

**Comment by RWTHLHK** [comment]
i find it in the source codes. It is called mechanical_strain.

**Reply by GiudGiud** [reply, marked as answer]
Other options that would have worked:
- output all material properties to exodus by adding output=exodus to the materials blocks
- Use the show_material_properties option in the Debug block

I m glad you had a look at the code though! Good reason to open source :)

**Reply by RWTHLHK** [reply]
Oh, got it! Thanks a lot!

**Reply by hugary1995** [reply]
I would use `elastic_strain` instead of `mechanical_strain`. Though if your simulation is purely elastic, they are equivalent.
